lib/_http_client.js

```diff
@@ -328,15 +328,19 @@ function socketCloseListener() {
   // NOTE: It's important to get parser here, because it could be freed by
   // the `socketOnData`.
   var parser = socket.parser;
-  if (req.res && req.res.readable) {
-    // Socket closed before we emitted 'end' below.
-    if (!req.res.complete) req.res.emit('aborted');
-    var res = req.res;
-    res.on('end', function() {
+  var res = req.res;
+  if (res) {
+    if (res.readable) {
+      // Socket closed before we emitted 'end' below.
+      res.emit('aborted');
```

Comment:
The `!req.res.complete` condition is gone, is this wanted?

Reply:
Is there any edit required here?

Reply:
It should have `if (!req.res.complete) req.res.emit('aborted');`. I would check the rest too. There have been changes to that part of code since that original PR was started.